engine/access/rpc/backend/backend_transactions.go

```diff
@@ -342,25 +347,31 @@ func (b *backendTransactions) GetTransactionResultsByBlockID(
 		}
 	}
 
-	// system chunk transactions
-	for i < len(resp.TransactionResults) {
-		txResult := resp.TransactionResults[i]
-		// tx body is irrelevant to status if it's in an executed block
-		txStatus, err := b.deriveTransactionStatus(nil, true, block)
-		if err != nil {
-			return nil, convertStorageError(err)
-		}
-
-		results = append(results, &access.TransactionResult{
-			Status:       txStatus,
-			StatusCode:   uint(txResult.GetStatusCode()),
-			Events:       convert.MessagesToEvents(txResult.GetEvents()),
-			ErrorMessage: txResult.GetErrorMessage(),
-			BlockID:      blockID,
-		})
+	// system chunk transaction
+	if i >= len(resp.TransactionResults) {
+		return nil, errInsufficientResults
+	} else if i < len(resp.TransactionResults)-1 {
+		return nil, status.Errorf(codes.Internal, "number of transaction results returned by execution node is more than the number of transactions in the block")
+	}
 
-		i++
+	systemTx, err := blueprints.SystemChunkTransaction(b.chainID.Chain())
+	if err != nil {
+		return nil, fmt.Errorf("could not get system chunk transaction: %w", err)
 	}
+	systemTxResult := resp.TransactionResults[len(resp.TransactionResults)-1]
+	systemTxStatus, err := b.deriveTransactionStatus(nil, true, block)
```

Comment:
really just being picky, but shouldn't we pass `systemTx` as a first arg here?

Reply:
True, I can do that